File: apps/site/public/reference/native/TPNativeReference.swift
// TP VPN native UI reference · iOS 17+ · integrate and compile in the target app.
// Inject real VPN state and callbacks. This view never creates a VPN tunnel.
import SwiftUI

enum TPConnectionState { case disconnected, connecting, connected, error }

struct TPNativeReference: View {
    let state: TPConnectionState
    let onConnect: () -> Void
    let onCancel: () -> Void
    let onDisconnect: () -> Void
    let onSelectNode: (String) -> Void
    @Environment(\.accessibilityReduceMotion) private var reduceMotion
    @State private var showNodes = false
    @State private var selectedNode = "Hong Kong"
    @State private var autoConnect = true
    private let brand = Color(red: 22 / 255, green: 119 / 255, blue: 1)
    private var label: String {
        switch state {
        case .disconnected: "Connect"
        case .connecting: "Cancel connection"
        case .connected: "Disconnect"
        case .error: "Retry connection"
        }
    }
    var body: some View {
        NavigationStack {
            List {
                Section {
                    Button {
                        switch state {
                        case .connecting: onCancel()
                        case .connected: onDisconnect()
                        default: onConnect()
                        }
                    } label: {
                        VStack(spacing: 16) {
                            Image(systemName: state == .connected ? "checkmark.shield.fill" : "power")
                                .font(.largeTitle).accessibilityHidden(true)
                            Text(label).font(.headline)
                            if state == .connecting { Text("Connecting…").font(.body) }
                            if state == .error { Text("Connection failed. Try again.").font(.body) }
                        }
                        .frame(maxWidth: .infinity, minHeight: 128).padding(24)
                        .foregroundStyle(brand)
                    }
                    .buttonStyle(.plain)
                    .animation(reduceMotion ? nil : .easeOut(duration: 0.18), value: state)
                }
                Section("Location") {
                    Button { showNodes = true } label: {
                        Label(selectedNode, systemImage: "globe").frame(minHeight: 44)
                    }
                }
                Section("Preferences") {
                    Toggle("Auto-connect", isOn: $autoConnect)
                }
            }
            .navigationTitle("TP VPN")
            .sheet(isPresented: $showNodes) {
                NavigationStack {
                    List(["Hong Kong", "Tokyo", "Singapore"], id: \.self) { node in
                        Button {
                            selectedNode = node
                            onSelectNode(node)
                            showNodes = false
                        } label: {
                            HStack {
                                Text(node)
                                Spacer()
                                if node == selectedNode { Image(systemName: "checkmark") }
                            }.frame(minHeight: 44)
                        }
                        .accessibilityAddTraits(node == selectedNode ? .isSelected : [])
                    }
                    .navigationTitle("Locations")
                    .toolbar { ToolbarItem(placement: .confirmationAction) { Button("Done") { showNodes = false } } }
                }
                .presentationDetents([.medium, .large])
                .presentationDragIndicator(.visible)
            }
        }
        .tint(brand)
        .preferredColorScheme(.light)
    }
}

File: apps/site/public/reference/native/TPNativeReference.kt
// TP VPN native UI reference · Compose Material 3.
// Compile in target app; inject VPN state, node selection and persisted settings.
package com.tpvpn.reference

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.*
import androidx.compose.ui.unit.dp

enum class TPConnectionState { Disconnected, Connecting, Connected, Error }

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TPNativeReference(
    state: TPConnectionState,
    onConnect: () -> Unit,
    onCancel: () -> Unit,
    onDisconnect: () -> Unit,
    onSelectNode: (String) -> Unit,
    autoConnect: Boolean,
    onAutoConnectChange: (Boolean) -> Unit
) {
    var showNodes by remember { mutableStateOf(false) }
    var selectedNode by remember { mutableStateOf("Hong Kong") }
    val label = when (state) {
        TPConnectionState.Disconnected -> "Connect"
        TPConnectionState.Connecting -> "Cancel connection"
        TPConnectionState.Connected -> "Disconnect"
        TPConnectionState.Error -> "Retry connection"
    }
    // Native component transitions honor the platform duration scale.
    // No decorative loop; a static text state remains when animation is disabled.
    MaterialTheme(colorScheme = lightColorScheme(
        primary = Color(0xFF1677FF), onPrimary = Color.White,
        background = Color(0xFFFAFBFF), surface = Color.White
    )) {
        Scaffold(topBar = { TopAppBar(title = { Text("TP VPN") }) }) { insets ->
            Column(Modifier.padding(insets).consumeWindowInsets(insets)
                .padding(24.dp).fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(24.dp)) {
                Text("A clearer connection", style = MaterialTheme.typography.headlineMedium)
                Button(
                    onClick = {
                        when (state) {
                            TPConnectionState.Connecting -> onCancel()
                            TPConnectionState.Connected -> onDisconnect()
                            else -> onConnect()
                        }
                    }, modifier = Modifier.fillMaxWidth().heightIn(min = 56.dp)
                ) { Text(label) }
                Text(when (state) {
                    TPConnectionState.Disconnected -> "Not connected"
                    TPConnectionState.Connecting -> "Connecting…"
                    TPConnectionState.Connected -> "Connected"
                    TPConnectionState.Error -> "Connection failed. Try again."
                }, modifier = Modifier.semantics { liveRegion = LiveRegionMode.Polite })
                OutlinedButton(onClick = { showNodes = true }, modifier = Modifier.heightIn(min = 48.dp)) {
                    Text(selectedNode)
                }
                ListItem(headlineContent = { Text("Auto-connect") }, trailingContent = {
                    Switch(checked = autoConnect, onCheckedChange = onAutoConnectChange,
                        modifier = Modifier.semantics { contentDescription = "Auto-connect" })
                })
            }
        }
        if (showNodes) {
            ModalBottomSheet(onDismissRequest = { showNodes = false }) {
                Column(Modifier.navigationBarsPadding().padding(24.dp), verticalArrangement = Arrangement.spacedBy(12.dp)) {
                    Text("Locations", style = MaterialTheme.typography.titleLarge)
                    listOf("Hong Kong", "Tokyo", "Singapore").forEach { node ->
                        TextButton(onClick = {
                            selectedNode = node
                            onSelectNode(node)
                            showNodes = false
                        }, modifier = Modifier.fillMaxWidth().heightIn(min = 48.dp)
                            .semantics { selected = node == selectedNode }) { Text(node) }
                    }
                }
            }
        }
    }
}
